### ISSUE TITLE
Less and Sass in style tab

### ISSUE THREAD

**Issue by 5e-Cleric** [naturalcrit/homebrewery#2793]
# Features
The style tab in the editor is great, but it can be better. Here are a bunch of suggestions as a tasklist to keep track of them if they ever get implemented

- [x] CSS rule folding
- [ ] Less and Sass compatibility

## Nesting features
And native CSS Nesting is here! all major PC browsers support it now.

So related to that i suggest even more features:

- [ ] Proper CSS Nesting highlight (broken now, probably bc of codemirror)
- [ ] Bracket pair tracking, either by color or by vertical lines (or both)

**Comment by ericscheid** [comment]
The CSS syntax highlighting is done via a codemirror thing .. and that's what will likely need updating. Any word on if codemirror are on to that?

**Comment by Gazook89** [comment]
I asked:

![image](https://github.com/naturalcrit/homebrewery/assets/58999374/4909e559-b7b6-44b4-81f2-42530d5394e0)

**Comment by dbolack-ab** [comment]
Are the nesting features implemented? I thought they were?

Was a decision made on the wisdom or lack of LESS and SASS in the style tab?

**Comment by 5e-Cleric** [comment]
> Are the nesting features implemented? I thought they were?

css has nesting natively, so the css nested will work, but the syntax is not correctly highlighted, correct syntax highlight for nesting is blocked by the codemirror update.

> Was a decision made on the wisdom or lack of LESS and SASS in the style tab?

There's a faint agreement that it could be beneficial, as long as we don't need to update it much, and there are examples of codemirror apps with less and sass syntax, (https://github.com/webextensions/live-css-editor), but it is a complex matter.

**Comment by Gazook89** [comment]
I vote we can close this.  I don't think LESS/SASS is very beneficial over just native CSS these days.  Nesting was the big one.  I believe there are even steps towards native mixins.  And a lot of new selectors in recent years.

Regarding outdated CSS syntax highlighting, per Marijn it isn't going to be updated.  Our next move towards fixing this would be covered by moving to CM6.

**Comment by 5e-Cleric** [comment]
I agree, closing in favor of #3511
